wasm/wasm.go

```diff
@@ -109,3 +134,27 @@ type Memory interface {
 	// Write writes the slice to the underlying buffer at the offset or returns false if out of range.
 	Write(offset uint32, v []byte) bool
 }
+
+// EncodeF32 converts the input so that it can be used as a Function F32 parameter or result.
+// See DecodeF32
+func EncodeF32(input float32) uint64 {
+	return uint64(math.Float32bits(input))
+}
+
+// DecodeF32 converts the Function F32 parameter or result to a float32.
+// See DecodeF32
+func DecodeF32(input uint64) float32 {
+	return float32(math.Float64frombits(input))
```

Comment:
yeah this seems wrong to me again, and should be `math.Float32frombits(uint32(input))` otherwise something must be wrong..

Reply:
https://github.com/tetratelabs/wazero/blob/cleaner-api/tests/spectest/spec_test.go#L79

Reply:
https://github.com/tetratelabs/wazero/blob/cleaner-api/tests/spectest/spec_test.go#L441-L442

Reply:
changing this breaks tests as the sign bit is lost, so I'll have to change this and also the encode side..

Reply:
this (`math.Float32frombits(uint32(input))`) itself shouldn't lose the signbit so something internal going wrong?

Reply:
ok found the flaw

Reply:
lemme

Reply:
https://github.com/tetratelabs/wazero/pull/268

Reply:
there was a flaw in engine :D